### PR TITLE
[Tezos] Implement block api from rpc

### LINKED ISSUE #1360: [Tezos] parsing staking related txs in BlockApi
RPC

- /chains/main/blocks/1253645
- /chains/main/blocks/1298011/context/contracts/tz1WDujRWCYjLBDfZieXW6insg5EUbg1rCRK

delegate (`amount` is empty): 

```json
{
  "kind": "delegation",
  "source": "tz1WDujRWCYjLBDfZieXW6insg5EUbg1rCRK",
  "fee": "1500",
  "counter": "2409159",
  "gas_limit": "10600",
  "storage_limit": "257",
  "delegate": "tz1TDSmoZXwVevLTEvKCTHWpomG76oC9S2fJ",
  "metadata": {
    "balance_updates": [
      {
        "kind": "contract",
        "contract": "tz1WDujRWCYjLBDfZieXW6insg5EUbg1rCRK",
        "change": "-1500"
      },
      {
        "kind": "freezer",
        "category": "fees",
        "delegate": "tz1gfArv665EUkSg2ojMBzcbfwuPxAvqPvjo",
        "cycle": 306,
        "change": "1500"
      }
    ],
    "operation_result": {
      "status": "applied",
      "consumed_gas": "1000",
      "consumed_milligas": "1000000"
    }
  }
}
```

undelegate (`delegate` is empty):

```json
{
  "kind": "delegation",
  "source": "tz1WDujRWCYjLBDfZieXW6insg5EUbg1rCRK",
  "fee": "1500",
  "counter": "2409162",
  "gas_limit": "10600",
  "storage_limit": "257",
  "metadata": {
    "balance_updates": [
      {
        "kind": "contract",
        "contract": "tz1WDujRWCYjLBDfZieXW6insg5EUbg1rCRK",
        "change": "-1500"
      },
      {
        "kind": "freezer",
        "category": "fees",
        "delegate": "tz1gfArv665EUkSg2ojMBzcbfwuPxAvqPvjo",
        "cycle": 316,
        "change": "1500"
      }
    ],
    "operation_result": {
      "status": "applied",
      "consumed_gas": "1000",
      "consumed_milligas": "1000000"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by hewigovens**
Continue of https://github.com/trustwallet/blockatlas/pull/1353, fixes #1360

### CODE REVIEW

File: platform/tezos/block.go [hunk 1]
```diff
@@ -1,20 +1,117 @@
...
-	return &types.Block{
-		Number: num,
-		Txs:    txs,
+	return NormalizeRpcBlock(block)
+}
+
+func NormalizeRpcBlock(block RpcBlock) (*types.Block, error) {
+	txs := []types.Tx{}
+
+	for _, ops := range block.Operations {
```
Comment (madcake) [inline platform/tezos/block.go:28]: Could we simplify it?

**Review by madcake — APPROVED** [review]

File: platform/tezos/block.go (hunk 1 above)
Comment (hewigovens) [reply platform/tezos/block.go:28]: unfortunately no, if you look at the json response

File: platform/tezos/block.go [hunk 2]
```diff
@@ -1,20 +1,117 @@
...
+	return &types.Block{Number: block.Header.Level, Txs: txs}, nil
+}
+
+func NormalizeRpcTransaction(tx RpcTransaction, header RpcBlockHeader) (types.Tx, error) {
+	coin := coin.Tezos()
+
+	var metadata interface{}
+	var to string
+	var txType types.TransactionType
+	if tx.Kind == TxTypeTransaction {
```
Comment (vikmeup) [inline platform/tezos/block.go:50]: can you use switch?

**Review by vikmeup — APPROVED** [review]

File: platform/tezos/block.go (hunk 2 above)
Comment (hewigovens) [reply platform/tezos/block.go]: done

**Pull request merged**